build/do.rq

```diff
@@ -1,51 +1,202 @@
 #!/usr/bin/env rq
 # rq: query data.script.main
-# rq: output null
+# rq: output /dev/null
+
+# METADATA
+# title: Regal build tasks
+# description: |
+#   This is the `rq` build tasks used by the Regal project. Note that `rq` is possibly not
+#   intended as a build tool, but if we can lint Rego with Rego, surely we can build Regal
+#   with Rego too?
+#
+#   To run one or more tasks, make sure to first have `rq` installed
+#   (https://git.sr.ht/~charles/rq), then from the project root directory, run:
+#
+#   build/do.rq <tasks>
+#
+#   For example, to run the `test` task, run:
+#
+#   build/do.rq test
+#
+# authors:
+#   - https://github.com/StyraInc/regal/graphs/contributors
+# related_resources:
+#   - description: rq
+#     ref: https://git.sr.ht/~charles/rq
+package script
+
 import future.keywords
 
-what := rq.args()[0]
-main := do[what] { true }
-else := job[what]
+main contains do[what]  if some what in rq.args()
+main contains job[what] if some what in rq.args()
+main contains job.tasks if {
+	count(rq.args()) == 0
+	print("No task(s) provided. Available tasks:")
+	print()
+}
+
+main contains null if {
+	count(rq.args()) == 1
+	rq.args()[0] == "--help"
+
+	print(rego.metadata.chain()[1].annotations.description)
+}
 
+# METADATA
+# title: pull_request
+# description: Run all task to verify a pull request
 do.pull_request {
 	some x in ["test", "lint", "e2e", "check_readme"]
 	github("::group::", x)
 	job[x]
 	github("::endgroup::", x)
 }
 
+# METADATA
+# title: tasks
+# description: Prints the name of all available tasks
+job.tasks {
+	some task in tasks
+	print("-", sprintf("%-20s", [task[0]]), "\t", strings.replace_n({"\n": ""}, task[1]))
+}
+
+# METADATA
+# title: pr
+# description: |
+#   Run all recommended tasks before submitting a PR
+# related_resources:
+#   - https://github.com/daixiang0/gci
+#   - https://github.com/mvdan/gofumpt
+#   - https://github.com/golangci/golangci-lint
+#   - https://github.com/open-policy-agent/opa
 job.pr {
-	print("thanks for contributing")
+	# format
+	fmt_all
+	write_readme
+
+	# verify
+	golangcilint
+	lint
+	test
+	e2e
 }
 
+# METADATA
+# title: test
+# description: Run all Regal unit tests (Go and Rego)
 job.test {
-	run("go test ./...")
+	test
 }
 
+# METADATA
+# title: lint
+# description: Run `regal lint` on the Regal bundle
 job.lint {
 	build
-	run("./regal lint --format github bundle")
+	lint_ci
 }
 
+# METADATA
+# title: e2e
+# description: Run the Regal end-to-end tests
 job.e2e {
 	build
-	run("go test -tags e2e ./e2e")
+	e2e
 }
 
+# METADATA
+# title: check_readme
+# description: Verify that the rules table in the README is up-to-date
 job.check_readme {
 	build
-	run("./regal table --compare-to-readme bundle")
+	check_readme
+}
+
+# METADATA
+# title: fetch_metadata
+# description: Fetch the built-in metadata definitions and store them in the bundle
+job.fetch_metadata {
+	out := rq.run([
+		"opa", "eval",
```

Comment:
I was about to say "just do rego.eval" but I start to loose track of our OPA-ish things. `rego.Eval` was EOPA, not rq. But `rq` could totally use a builtin like that, too.

Reply:
I was thinking the same thing!